metro/proto/v1/spec.proto

```diff
@@ -528,6 +528,15 @@ message PushConfig {
         string audience = 2;
     }
 
+    // Implementation of authentication_method based on Basic Auth 
+    // which uses username and password for authentication
+    message BasicAuth {
```

Comment:
Might as well design with an extension in mind? : e.g. applications in the future will start accepting JWTs instead of basic auth.

Reply:
I think in that case we would be creating another implementation for the same. Idea is to provide implementations for all the auth methods that we will be supporting in future. But for that, we will have to re-design subscription model a bit to make it extensible for other auth methods. As discussed, we will be taking up that task separately in future.

Reply:
Sure. Go ahead. I may be missing context due to my lack of attendance in the recent meetings.

My suggestion was around **future-proofing** this `message` construct itself by maybe naming this as `Auth` and then have different fields within this as needed with an additional `identifier` which can take values as `basic-auth/jwt` in the future.